pkg/payload/precondition/clusterversion/recommendedupdate.go

```diff
@@ -0,0 +1,120 @@
+package clusterversion
+
+import (
+	"context"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/cluster-version-operator/lib/resourcemerge"
+	precondition "github.com/openshift/cluster-version-operator/pkg/payload/precondition"
+)
+
+// RecommendedUpdate checks if clusterversion is upgradeable currently.
+type RecommendedUpdate struct {
+	key    string
+	lister configv1listers.ClusterVersionLister
+}
+
+// NewRecommendedUpdate returns a new RecommendedUpdate precondition check.
+func NewRecommendedUpdate(lister configv1listers.ClusterVersionLister) *RecommendedUpdate {
+	return &RecommendedUpdate{
+		key:    "version",
+		lister: lister,
+	}
+}
+
+// Run runs the RecommendedUpdate precondition.
+// Returns PreconditionError when possible, if the requested target release is Recommended=False.
+func (ru *RecommendedUpdate) Run(ctx context.Context, releaseContext precondition.ReleaseContext) error {
+	clusterVersion, err := ru.lister.Get(ru.key)
+	if apierrors.IsNotFound(err) || meta.IsNoMatchError(err) {
+		return nil
```

Comment:
These seem like they could have something similar to the `Configured channel is unset...` error.  Although in practice, if there is no ClusterVersion here, we're unlikely to be able to store our complaint anywhere persistent, so :shrug:, maybe not worth trying to cover the "ClusterVersion doesn't exist now, but it will be created within a few ms in time for us to record this complaint" hole.